### ISSUE TITLE
[1.10.2] Crash when placing upgrade into transfer node AFTER updating to last version

### ISSUE THREAD

**Issue by kaal-1** [rwtema/extrautilities_old_issues#1669]
Description: Updating screen events

java.lang.NullPointerException: Updating screen events
	at com.rwtema.extrautils2.transfernodes.TransferNodeBase$1.getStackLimit(TransferNodeBase.java:59)
	at net.minecraftforge.items.ItemStackHandler.insertItem(ItemStackHandler.java:85)
	at net.minecraftforge.items.SlotItemHandler.func_75214_a(SlotItemHandler.java:56)
	at com.rwtema.extrautils2.gui.backend.DynamicContainer.func_75135_a(DynamicContainer.java:426)
	at com.rwtema.extrautils2.gui.backend.DynamicContainer.func_82846_b(DynamicContainer.java:357)
	at net.minecraft.inventory.Container.func_184996_a(Container.java:258)
	at com.rwtema.extrautils2.gui.backend.DynamicContainer.func_184996_a(DynamicContainer.java:492)
	at net.minecraft.client.multiplayer.PlayerControllerMP.func_187098_a(PlayerControllerMP.java:550)
	at net.minecraft.client.gui.inventory.GuiContainer.func_184098_a(GuiContainer.java:630)
	at net.minecraft.client.gui.inventory.GuiContainer.func_73864_a(GuiContainer.java:389)
	at com.rwtema.extrautils2.gui.backend.DynamicGui.func_73864_a(DynamicGui.java:321)
	at net.minecraft.client.gui.GuiScreen.func_146274_d(GuiScreen.java:548)
	at com.rwtema.extrautils2.gui.backend.DynamicGui.func_146274_d(DynamicGui.java:294)
	at net.minecraft.client.gui.GuiScreen.func_146269_k(GuiScreen.java:517)
	at net.minecraft.client.Minecraft.func_71407_l(Minecraft.java:1700)
	at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:1055)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:371)
	at net.minecraft.client.main.Main.main(SourceFile:124)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)

A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Client thread
Stacktrace:
	at com.rwtema.extrautils2.transfernodes.TransferNodeBase$1.getStackLimit(TransferNodeBase.java:59)
	at net.minecraftforge.items.ItemStackHandler.insertItem(ItemStackHandler.java:85)
	at net.minecraftforge.items.SlotItemHandler.func_75214_a(SlotItemHandler.java:56)
	at com.rwtema.extrautils2.gui.backend.DynamicContainer.func_75135_a(DynamicContainer.java:426)
	at com.rwtema.extrautils2.gui.backend.DynamicContainer.func_82846_b(DynamicContainer.java:357)
	at net.minecraft.inventory.Container.func_184996_a(Container.java:258)
	at com.rwtema.extrautils2.gui.backend.DynamicContainer.func_184996_a(DynamicContainer.java:492)
	at net.minecraft.client.multiplayer.PlayerControllerMP.func_187098_a(PlayerControllerMP.java:550)
	at net.minecraft.client.gui.inventory.GuiContainer.func_184098_a(GuiContainer.java:630)
	at net.minecraft.client.gui.inventory.GuiContainer.func_73864_a(GuiContainer.java:389)
	at com.rwtema.extrautils2.gui.backend.DynamicGui.func_73864_a(DynamicGui.java:321)
	at net.minecraft.client.gui.GuiScreen.func_146274_d(GuiScreen.java:548)
	at com.rwtema.extrautils2.gui.backend.DynamicGui.func_146274_d(DynamicGui.java:294)
	at net.minecraft.client.gui.GuiScreen.func_146269_k(GuiScreen.java:517)

**Comment by rgtexa** [comment]
Also experiencing this same crash after updating SkyFactory 3 to the latest version.

Crash happens when trying to place an Upgrade Mining into a Liquid Transfer Node. Haven't tested if this impacts all upgrades and all nodes.

https://gist.github.com/rgtexa/43aca0cb0d73f8e8603d565a58d7d6a9

Extra Utilities 2: 1.5.2
Forge: 2422

**Comment by depressed-pho** [comment]
Yes exactly the same crash happens on extrautils2-1.10.2-1.5.2.jar, 100% reproducible with OptiFine installed. I don't know if it also happens without OptiFine though (for being too lazy to test. Sorry)

**Comment by Dylanavo** [comment]
I've also experienced this on extrautils2-1.10.2-1.5.2.jar

https://gist.github.com/Dylanavo/757fcb25fc56886f64f2c8b80e2a9f00

**Comment by Mason11987** [comment]
I get this same crash, don't have optifine.

**Comment by thelapisfox** [comment]
Same crash here as well but on version 1.6 of eu2. I do not have optifine.


EDIT: Error does not occur in 1.4.4

**Comment by melbellrocks** [comment]
Add us to the list getting this crash.  It happens when putting any upgrade into an item transfer node (haven't tried fluid nodes).  Playing Foolcraft 2 (EU2 version 1.5.2)

**Comment by wpmidia** [comment]
Same error here!

```

---- Minecraft Crash Report ----

WARNING: coremods are present:
  MicdoodlePlugin (MicdoodleCore-1.10.2-4.0.0.117.jar)
  BookshelfLoadingPlugin (Bookshelf-1.10.2-1.4.4.347.jar)
  CCLCorePlugin (CodeChickenLib-1.10.2-2.5.9.283-universal.jar)
  CTMCorePlugin (CTM-MC1.10.2-0.2.2.24.jar)
  CoFH Loading Plugin (CoFHCore-1.10.2-4.1.11.16-universal.jar)
  NEICorePlugin (NotEnoughItems-1.10.2-2.1.3.220-universal.jar)
  OpenModsCorePlugin (OpenModsLib-1.10.2-0.11.1.jar)
  MalisisCorePlugin (malisiscore-1.10.2-4.4.0.jar)
  EnderCorePlugin (EnderCore-1.10.2-0.4.1.66-beta.jar)
  Do not report to Forge! Remove FoamFixAPI (or replace with FoamFixAPI-Lawful) and try again. (foamfix-0.7.2-anarchy.jar)
  AppEngLoadingPlugin (appliedenergistics2-rv4-stable-1.jar)
  FMLPlugin (InventoryTweaks-1.61-58.jar)
Contact their authors BEFORE contacting forge

// I blame Dinnerbone.

Time: 9/11/17 3:37 PM
Description: Updating screen events

java.lang.NullPointerException: Updating screen events
	at com.rwtema.extrautils2.transfernodes.TransferNodeBase$1.getStackLimit(TransferNodeBase.java:59)
	at net.minecraftforge.items.ItemStackHandler.insertItem(ItemStackHandler.java:85)
	at net.minecraftforge.items.SlotItemHandler.func_75214_a(SlotItemHandler.java:56)
	at com.rwtema.extrautils2.gui.backend.DynamicContainer.func_75135_a(DynamicContainer.java:426)
	at com.rwtema.extrautils2.gui.backend.DynamicContainer.func_82846_b(DynamicContainer.java:357)
	at net.minecraft.inventory.Container.func_184996_a(Container.java:258)
	at com.rwtema.extrautils2.gui.backend.DynamicContainer.func_184996_a(DynamicContainer.java:492)
	at net.minecraft.client.multiplayer.PlayerControllerMP.func_187098_a(PlayerControllerMP.java:550)
	at codechicken.nei.guihook.GuiContainerManager.handleSlotClick(GuiContainerManager.java:557)
	at net.minecraft.client.gui.inventory.GuiContainer.func_184098_a(GuiContainer.java:627)
	at net.minecraft.client.gui.inventory.GuiContainer.public_func_184098_a(GuiContainer.java)
	at codechicken.nei.guihook.DefaultSlotClickHandler.callHandleMouseClick(DefaultSlotClickHandler.java)
	at codechicken.nei.guihook.DefaultSlotClickHandler.handleSlotClick(DefaultSlotClickHandler.java:15)
	at codechicken.nei.guihook.GuiContainerManager.handleMouseClick(GuiContainerManager.java:522)
	at net.minecraft.client.gui.inventory.GuiContainer.managerHandleMouseClick(GuiContainer.java)
	at net.minecraft.client.gui.inventory.GuiContainer.func_73864_a(GuiContainer.java:389)
	at com.rwtema.extrautils2.gui.backend.DynamicGui.func_73864_a(DynamicGui.java:321)
	at net.minecraft.client.gui.GuiScreen.func_146274_d(GuiScreen.java:548)
	at com.rwtema.extrautils2.gui.backend.DynamicGui.func_146274_d(DynamicGui.java:294)
	at net.minecraft.client.gui.GuiScreen.func_146269_k(GuiScreen.java:517)
	at net.minecraft.client.Minecraft.func_71407_l(Minecraft.java:1700)
	at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:1055)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:371)
	at net.minecraft.client.main.Main.main(SourceFile:124)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:236)
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:297)
	at org.multimc.EntryPoint.listen(EntryPoint.java:162)
	at org.multimc.EntryPoint.main(EntryPoint.java:53)
```


...

**Comment by tektx** [comment]
I also experienced a crash when putting an Upgrade Mining in a Fluid Transport Node on extrautils2-1.10.2-1.5.2.jar. The issue was resolved after upgrading to extrautils2-1.10.2-1.6.1.jar.

**Comment by mpavkovic** [comment]
If you're playing Sky Factory 3, until modpack maker updates the pack, you can manually update Extra Utilities to latest version (I can confirm that 1.6.3 doesn't cause crashes)
